### ISSUE TITLE
Are customized loss functions/layers supported?

### ISSUE THREAD

**Issue by zzpustc** [AlexImmer/Laplace#119]
Hi!

Thanks for the great work. I'm trying to use the Laplace approximation in my work, but your package only supports MSE and CE, and the corresponding layers only support the nn.Module class. Is there any method to use your package with customized loss functions/layers?

Best

**Comment by AlexImmer** [comment]
Hi,

Thanks for your interest. Unfortunately, only basic nn.Modules can be supported since this allows to compute Hessian approximations (for example see the necessary extensions [in ASDL](https://github.com/kazukiosawa/asdl/tree/master/asdfghjkl/operations)). The same applies to losses. 

However, in some cases it is not that complicated to extend the corresponding backend so if you have a specific use-case, we can try to give suggestions how to get it done with the help of the library if possible.

**Comment by wiseodd** [comment]
The `CurvlinopsGGN/EF` with the diagonal structure should be able to handle non-module layers (as long as its params are included in `model.parameters()`) since it's just a pure `torch.func`.

For custom loss functions, the requirement is that they correspond to a log-likelihood. Then we need to know how to sample or compute the 2nd derivative w.r.t. network's output. E.g. here https://github.com/aleximmer/Laplace/blob/7541c9bec28691b55352ac9895e92a4e351f6fc8/laplace/curvature/curvature.py#L239-L257

We plan to support more likelihood, e.g. BCE (#130), after milestone 0.2.
